client/pages/cart.js

```diff
@@ -1,71 +1,88 @@
 import React, { useEffect, useState } from "react";
-import { Row, Col, ListGroup, Button, Card, Container } from "react-bootstrap";
+import {
+  Row,
+  Col,
+  ListGroup,
+  Button,
+  Card,
+  Container,
+  Form,
+} from "react-bootstrap";
 import Router from "next/router";
 import Link from "next/link";
 
 import Message from "../components/Message";
 import CheckoutSteps from "../components/CheckoutSteps";
 import NextImage from "../components/NextImage";
+import buildClient from "../api/build-client";
 
-const CartPage = ({ currentUser }) => {
+const CartPage = ({ currentUser, products }) => {
+  const [cart, setCart] = useState(null);
+  const [color, setColor] = useState([]);
+  const [selectedColor, setSelectedColor] = useState(null);
+  const [size, setSize] = useState([]);
+  const [selectedSize, setSelectedSize] = useState(null);
   const [productId, setProductId] = useState(null);
-  const [onIncrease, setOnIncrease] = useState(false);
-  const [onDecrease, setOnDecrease] = useState(false);
   const [deletedItemId, setDeletedItemId] = useState(null);
 
   const [storageReady, setStorageReady] = useState(false);
-  const [cart, setCart] = useState(null);
   const [onEdit, setOnEdit] = useState(false);
+  const [onIncrease, setOnIncrease] = useState(false);
+  const [onDecrease, setOnDecrease] = useState(false);
   const [onRemove, setOnRemove] = useState(false);
-  const [hasOption, setHasOption] = useState(false);
-  // const [hasSize, sethasSize] = useState(false);
 
   useEffect(() => {
+    // Get cartItems from LocalStorage
     const cartItems = localStorage.getItem("cartItems")
       ? JSON.parse(localStorage.getItem("cartItems"))
       : [];
 
-    // Cart has items or empty
-    if (cartItems !== undefined) {
+    // cartItems haves items
+    if (cartItems !== undefined && cartItems !== null) {
```

Comment:
need to check whether it is null or undefined when getting an empty `cartItems` from `localStorage `